src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs

```diff
@@ -38,10 +111,168 @@ private void SetContentDispositionHeader(ActionContext context, FileResult resul
             }
         }
 
-        private void SetContentType(ActionContext context, FileResult result)
+        private static void SetAcceptRangeHeader(ActionContext context)
         {
             var response = context.HttpContext.Response;
-            response.ContentType = result.ContentType;
+            response.Headers[HeaderNames.AcceptRanges] = AcceptRangeHeaderValue;
+        }
+
+        // Internal for testing
+        internal static PreconditionState GetPreconditionState(
+            ActionContext context,
+            RequestHeaders httpRequestHeaders,
+            DateTimeOffset? lastModified = null,
+            EntityTagHeaderValue etag = null)
+        {
+            var ifMatchState = PreconditionState.Unspecified;
+            var ifNoneMatchState = PreconditionState.Unspecified;
+            var ifModifiedSinceState = PreconditionState.Unspecified;
+            var ifUnmodifiedSinceState = PreconditionState.Unspecified;
+
+            // 14.24 If-Match
+            var ifMatch = httpRequestHeaders.IfMatch;
+            if (ifMatch != null && ifMatch.Any())
```

Comment:
You can extract this block of code into a method and reuse it below, something like:
EtagMatches(ifMatch,etag,matchingResult,notMatchingResult){
if (ifMatch != null && ifMatch.Any())
{
    ifMatchState = notMachingResult;
    foreach (var entityTag in ifMatch)
    {
        if (entityTag.Equals(EntityTagHeaderValue.Any) || entityTag.Compare(etag, useStrongComparison: true))
        {
            ifMatchState = matchingResult;
            break;
        }
    }
}
}

The same technique can be applied to the ifModifiedSince